src/backend/bs_repserver

```diff
@@ -3154,24 +3154,33 @@ sub getjobhistory {
   if ($cgi->{'code'} && @{$cgi->{'code'}} == 1 && $cgi->{'code'}->[0] eq 'lastfailures') {
     return getlastfailures($cgi, $projid, $repoid, $arch);
   }
+  my $idx = 0;
+  my %indices = map {$_ => $idx++} @$BSXML::jobhistlay;
+  my $packidx = $indices{'package'};
+  my $codeidx = $indices{'code'};
+  my $endtimeidx = $indices{'endtime'};
   if ($cgi->{'package'} && $cgi->{'code'}) {
-    my %packid = map {$_ => 1} @{$cgi->{'package'}};
+    my %packid = map {BSFileDB::encode_field($_) => 1} @{$cgi->{'package'}};
     my %code = map {$_ => 1} @{$cgi->{'code'}};
-    $filter = sub {$packid{$_[0]->{'package'}} && $code{$_[0]->{'code'}}};
+    $filter = sub {$packid{$_[$packidx]} && $code{$_[$codeidx]}};
   } elsif ($cgi->{'package'}) {
-    my %packid = map {$_ => 1} @{$cgi->{'package'}};
-    $filter = sub {$packid{$_[0]->{'package'}}};
+    my %packid = map {BSFileDB::encode_field($_) => 1} @{$cgi->{'package'}};
+    $filter = sub {$packid{$_[$packidx]}};
```

Comment:
I still don't get how this can work if the filter is called with the complete string. Don't you want to split the line into fields before calling the filter function?

Reply:
I might be missing something. 
But I have added `my @line = split('\|', $_);` to the endtime filter (couple of lines below).
Then `@line` gets passed to other filters

Reply:
But only if the endtime filter is used...

Oh wait, there's a bogus `|| 1` in the code! A leftover from some testing, I think.

So your code works, but only because of that bug...

Reply:
Anyway, it's IMHO bad style. I think the BSFileDB code should do the split if $lay is undef.